pkgs/development/python-modules/fluidasserts/default.nix

```diff
@@ -0,0 +1,142 @@
+{ buildPythonPackage
+, fetchPypi
+, isPy37
+, lib
+
+# pythonPackages
+, aiohttp
+, androguard
+, azure-identity
+, azure-keyvault-keys
+, azure-keyvault-secrets
+, bandit
+, bcrypt
+, beautifulsoup4
+, boto3
+, cfn-flip
+, cython
+, dnspython
+, colorama
+, configobj
+, defusedxml
+, GitPython
+, google_api_python_client
+, kubernetes
+, ldap3
+, mysql-connector
+, names
+, ntplib
+, oyaml
+, paramiko
+, pillow
+, psycopg2
+, pycrypto
+, pygments
+, pyjks
+, pynacl
+, pyopenssl
+, pypdf2
+, pysmb
+, python_magic
+, pytz
+, requirements-detector
+, selenium
+, tlslite-ng
+, viewstate
+
+# custom pythonPackages
+, mixpanel'
+}:
+
+buildPythonPackage rec {
+  pname = "fluidasserts";
+  version = "20.1.21324";
+  disabled = !isPy37;
+
+  src = fetchPypi {
+    inherit pname version;
+    format = "setuptools";
+    extension = "zip";
+    sha256 = "061jhd9s3mikzld8240r121clcdxsyniqbjqcripii387khq4jxz";
+  };
+
+  patchPhase = ''
+    # Version mismatches between current FluidAsserts and Nixpkgs
+    substituteInPlace ./setup.py \
+      --replace 'pycrypto==2.6.1' 'pycrypto==3.9.4' \
+      --replace 'tlslite-ng==0.8.0-alpha29' 'tlslite-ng==0.7.5' \
+      --replace 'boto3==1.10.17' 'boto3==1.10.1' \
+      --replace 'cfn-flip==1.2.2' 'cfn-flip==1.1.0.post1' \
+
+    # Functionality that will be not present for the momment
+    #   but that we'll work to add in the future
+    substituteInPlace ./setup.py \
+      --replace "'azure==4.0.0'," "" \
+      --replace "'pytesseract==0.3.0'," "" \
+      --replace "'pymssql==2.1.4'," "" \
+      --replace "'pywinrm==0.4.1'," "" \
+      --replace "'azure-storage-file-share==12.0.0'," "" \
+
+  '';
+
+  propagatedBuildInputs = [
+    # pythonPackages
+    aiohttp
+    androguard
+    azure-identity
+    azure-keyvault-keys
+    azure-keyvault-secrets
+    bandit
+    bcrypt
+    beautifulsoup4
+    boto3
+    cfn-flip
+    cython
+    dnspython
+    colorama
+    configobj
+    defusedxml
+    GitPython
+    google_api_python_client
+    kubernetes
+    ldap3
+    mysql-connector
+    names
+    ntplib
+    oyaml
+    paramiko
+    pillow
+    psycopg2
+    pycrypto
+    pygments
+    pyjks
+    pynacl
+    pyopenssl
+    pypdf2
+    pysmb
+    python_magic
+    pytz
+    requirements-detector
+    selenium
+    tlslite-ng
+    viewstate
+
+    # custom pythonPackages
+    mixpanel'
+  ];
+
+  # Tests are not shipped via PyPi, we'll work on it
+  #   https://gitlab.com/fluidattacks/asserts/issues/869
+  #
+  # Manual testing has been conducted and it works fine
+  doCheck = false;
```

Comment:
is it feasible to check out the git repo?

Reply:
yes, it's feasible, however there are two points:
- the majority of checks assume network connectivity (mainly to launch 20+ docker containers)
- for some reason I coudn't pass the 'setup phase' when installing from source code

I think i'll just ship the tests in PyPi to test at least the ones that can be tested without network connectivity